pkg/server/util/regentryutil/fetch.go

```diff
@@ -25,34 +28,43 @@ func newRegistrationEntryFetcher(dataStore datastore.DataStore) *registrationEnt
 	}
 }
 
-func (f *registrationEntryFetcher) Fetch(ctx context.Context, id string) ([]*common.RegistrationEntry, error) {
-	entries, err := f.fetch(ctx, id, make(map[string]bool))
+func (f *registrationEntryFetcher) Fetch(ctx context.Context, cache RegistrationEntriesCache, id string) ([]*common.RegistrationEntry, error) {
```

Comment:
Since the TTL is smaller than the agent sync interval, it seems like entries cached for the agent SPIFFE ID will always be expired by the time the cache entry is retrieved. Therefore, I think our cache can be much lighter if we do not cache the entries for the agent SPIFFE ID (i.e. the first call to `fetch`).

Reply:
sure added `shouldCache` variable in implementation to just cache recursive calls